hack/build-coredns.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+PLUGIN_PATH=$(readlink -f "$(dirname "$0")/..")
+
+# Create a temporary directory for cloning openshift/coredns repo.
+# The directory will be deleted after the execution of script.
+BASE_PATH=$(mktemp -d)
+trap "chmod -R u+w $BASE_PATH; rm -rf $BASE_PATH" EXIT
+
+CURRENT_DIR=$(pwd)
+
+cd $BASE_PATH
+# Clone openshift/coredns repo if not already cloned.
+if [ ! -d coredns ]
+then
+    git clone https://github.com/openshift/coredns
+fi
+
+# Add the "ocp_dnsnameresolver" plugin to the cloned openshift/coredns repo.
+GOFLAGS=-mod=mod $CURRENT_DIR/hack/add-plugin.sh $BASE_PATH/coredns $PLUGIN_PATH
```

Comment:
I think we don't need `$CURRENT_DIR` variable, it was used before for the last change directory `(cd ${CURRENT_DIR}`).
```suggestion
GOFLAGS=-mod=mod ${PLUGIN_PATH}/hack/add-plugin.sh $BASE_PATH/coredns $PLUGIN_PATH
```

Reply:
Btw it would be good to check the shell scripts with [the shellcheck](https://github.com/koalaman/shellcheck). Things like this could be eliminated:
```suggestion
GOFLAGS=-mod=mod $CURRENT_DIR/hack/add-plugin.sh "${BASE_PATH}/coredns" "${PLUGIN_PATH}"
```